README.md

```diff
@@ -12,62 +12,370 @@ Install from NPM:
+
+it('should wait', async() => {
+  await aTimeout(100); // fulfills after 100ms
+  await nextFrame(); // fulfills before next paint
+});
+```
+
+#### Waiting For Asynchronous Components
+
+`fixture()` will automatically wait for all nested Lit components to fulfill their `updateComplete` Promise. To tweak when a component's `updateComplete` fulfills, implement the [getUpdateComplete() lifecycle callback](https://lit.dev/docs/components/lifecycle/#getUpdateComplete).
+
+In other scenarios, a component may have an initial loading state (e.g. loading spinner or skeleton) where `updateComplete` has already resolved in addition to another fully loaded state. To signal that `fixture()` should wait for this final state, implement `getLoadingComplete()`. It works the same way as `getUpdateComplete()` by fulfilling its Promise when the component has fully loaded.
+
+```javascript
+class SlowElem extends LitElement {
+  render() {
+    return html`<p>I take my time</p>`;
+  }
+  async getLoadingComplete() {
+    return new Promise(resolve => {
+      setTimeout(() => resolve(), 2000);
+    });
+  }
+}
+```
+
+To bypass waiting for `getLoadingComplete()`, set the `awaitLoadingComplete` configuration option to `false`:
+
+```javascript
+it('should not wait', async() => {
+  const elem = await fixture(html`<slow-elem></slow-elem>`, {
+    awaitLoadingComplete: false
+  });
+});
+```
+
+#### Waiting for a Condition
+
+In cases where there are no other reliable hooks (like events, `getUpdateComplete()` or `getLoadingComplete()`), `waitUntil(condition)` can be used to wait for a particular condition to become `true`. The condition can optionally return a Promise.
+
+```javascript
+import { fixture, waitUntil } from '@brightspace-ui/testing';
+
+it('should wait for condition', async() => {
+  const elem = await fixture(...);
+  await waitUntil(() => elem.foo === 'bar');
+});
+```
+
+By default, `waitUntil` will poll every `50ms` and time out after `1000ms`. Those options can be configured:
+
+```javascript
+await waitUntil(() => elem.condition, {
+  interval: 10,
+  timeout: 2000
+});
+```
+
+> **Note:** because `waitUntil` constantly polls, it can slow down test execution and should be avoided if possible.
+
+### Defining a Custom Element for Testing
+
+If a test requires a one-off custom element, define it using `defineCE` and pass the returned tag name in the call to `fixture()`.
+
+```javascript
+import { defineCE, fixture, html } from '@brightspace-ui/testing';
+
+const tag = defineCE(
+  class extends LitElement {
+    static properties = {
+      foo: { type: String }
+    };
+    render() {
+      return html`hello`;
+    }
+  }
+);
+
+it('should use custom element', async() => {
+  const foo = 'bar';
+  const elem = await fixture(html`<${tag} foo="${foo}"></${tag}>`);
+});
 ```
 
-## Versioning & Releasing
+> **Important:** `defineCE` is not performant and shouldn't be used outside of tests.
```

Comment:
```suggestion
> **Important:** `defineCE` is not performant and shouldn't be used outside of test files.
```

Reply:
Oh, or does this mean used the result of `defineCE` in tests?

Reply:
From the open-wc docs:
> This uses a workaround that's not performant for re-renders, which is fine for testing, but do NOT use this in production code.

Reply:
Ok, then I still with the suggestion above for clarity since `defineCE` is usually used outside of the literal `it` test, but always only in test files.